label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -140,6 +173,12 @@ public void deleteLabelApplicationRule(
     try {
       RequestContext requestContext = RequestContext.CURRENT.get();
       this.requestValidator.validateOrThrow(requestContext, request);
+      String labelApplicationRuleId = request.getId();
+      if (systemLabelApplicationRulesMap.containsKey(labelApplicationRuleId)) {
+        // Deleting a system label application rule is not allowed
```

Comment:
We generally allow users to delete default config. Why is this different?

Reply:
Can be done. However deleting a default rule will require a new config store to persist it. Will add it if need arises.

Reply:
will do it in separate PR